### PR TITLE
Add support for ImageTransform to ObjectDetectionRecordReader

### PR DESCRIPTION
**Pull request by saudet**
WORK IN PROGRESS: DO NOT MERGE.

Related to issue #417 

## What changes were proposed in this pull request?

Add ImageTransform.query(coordinates) method so that ObjectDetectionRecordReader can use it to transform coordinates.

## How was this patch tested?

New unit tests pass.

### CODE REVIEW

File: datavec-data/datavec-data-image/src/main/java/org/datavec/image/recordreader/objdetect/ObjectDetectionRecordReader.java [hunk 1]
```diff
@@ -71,6 +72,25 @@ public ObjectDetectionRecordReader(int height, int width, int channels, int grid
...
 
+    /**
+     *
+     * @param height        Height of the output images
+     * @param width         Width of the output images
+     * @param channels      Number of channels for the output images
+     * @param gridH         Grid/quantization size (along  height dimension) - Y axis
+     * @param gridW         Grid/quantization size (along  height dimension) - X axis
+     * @param labelProvider ImageObjectLabelProvider - used to look up which objects are in each image
+     * @param imageTransform ImageTransform - used to transform image and coordinates
```
Comment (AlexDBlack) [inline datavec-data/datavec-data-image/src/main/java/org/datavec/image/recordreader/objdetect/ObjectDetectionRecordReader.java]: Maybe note the "object is removed if new center is outside of transformed image bounds" thing

File: datavec-data/datavec-data-image/src/main/java/org/datavec/image/recordreader/objdetect/ObjectDetectionRecordReader.java [hunk 2]
```diff
@@ -147,6 +167,15 @@ public void initialize(InputSplit split) throws IOException {
 
                 //put the label data into the output label array
                 for (ImageObject io : objectsThisImg) {
+                    if (imageTransform != null) {
+                        float[] pts = imageTransform.query(io.getX1(), io.getY1(), io.getX2(), io.getY2());
+                        io = new ImageObject(Math.round(pts[0]), Math.round(pts[1]), Math.round(pts[2]), Math.round(pts[3]), io.getLabel());
+                        double cx = io.getXCenterPixels();
+                        double cy = io.getYCenterPixels();
+                        if (cx < 0 || cx >= oW || cy < 0 || cy >= oH) {
+                            continue;
```
Comment (AlexDBlack) [inline datavec-data/datavec-data-image/src/main/java/org/datavec/image/recordreader/objdetect/ObjectDetectionRecordReader.java]: 👍

File: datavec-data/datavec-data-image/src/main/java/org/datavec/image/transform/ColorConversionTransform.java [hunk 3]
```diff
@@ -93,5 +93,7 @@ public ImageWritable transform(ImageWritable image, Random random) {
         return new ImageWritable(converter.convert(result));
     }
 
-
+    public float[] query(float... coordinates) {
```
Comment (AlexDBlack) [inline datavec-data/datavec-data-image/src/main/java/org/datavec/image/transform/ColorConversionTransform.java]: Need ```@Override``` annotation on all the query methods

**Comment by saudet** [comment]
Thanks for the feedback! If there are no other issues, I'll be merging this. It also fixes #433.

**Pull request merged**